plugin/pkg/scheduler/core/equivalence_cache.go

```diff
@@ -18,29 +18,34 @@ package core
 
 import (
 	"hash/fnv"
-
-	"github.com/golang/groupcache/lru"
-
 	"sync"
 
+	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/kubernetes/pkg/api/v1"
 	hashutil "k8s.io/kubernetes/pkg/util/hash"
 	"k8s.io/kubernetes/plugin/pkg/scheduler/algorithm"
+
+	"github.com/golang/glog"
+	"github.com/golang/groupcache/lru"
 )
 
-// TODO(harryz) figure out the right number for this, 4096 may be too big
-const maxCacheEntries = 4096
+// we use predicate names as cache's key, its count is limited
+const maxCacheEntries = 100
```

Comment:
Ah, I see. The three level indexing is complex. So, we have N caches, each of which belong to one node and has maxCacheEntries entries (for predicates). We use a predicate key to find an entry in the cache. The entry in the cache is a map. That map is indexed with equivalence hash. This third level confused me.

We could have probably dropped predicates from the equation at the cost of losing precision on failure reasons and cache invalidations.